stdlib/public/core/String.swift

```diff
@@ -205,20 +205,20 @@ extension String {
   /// - Parameter encoding: describes the encoding in which the code units
   ///   should be interpreted.
   public init<Encoding: UnicodeEncoding>(
-    cString nulTerminatedCodeUnits: UnsafePointer<Encoding.CodeUnit>,
-    encoding: Encoding.Type) {
+    decodingCString nulTerminatedCodeUnits: UnsafePointer<Encoding.CodeUnit>,
```

Comment:
The thinking here was that using the label `encoding:` makes it look like you're specifying the encoding of the resulting String storage, and that effect would become more pronounced when/if we ever support strings with other encodings.  Also it matches `init(decoding:as:)`.  If we change this, we should change that too.

It's hard to know what's best here, which is a reason I asked for review.  I had a few mental hiccups when updating the code to use this API but had to remind myself that how the code reads matters more than what it's like to write it.